packages/ember-runtime/lib/ext/string.js

```diff
@@ -14,9 +14,27 @@ import {
   capitalize,
   classify
 } from '../system/string';
+import { deprecate } from 'ember-debug';
+import { assign } from 'ember-utils';
 
 const StringPrototype = String.prototype;
 
+function deprecateEmberStringExtension(fn, opts = {}) {
+  let { name } = fn;
```

Comment:
I don't think we can rely on `Function.name` (AFAIK it isn't supported on all platforms, which would make this fail in production).

Checkout https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name for a decent compat table (it isn't supported on IE at all)...